pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -255,6 +255,8 @@ public PinotHelixResourceManager(ControllerConf controllerConf) {
    * as PARTICIPANT,
    * which would be put to lead controller resource and mess up the leadership assignment. Those places should use
    * SPECTATOR other than PARTICIPANT.
+   * TODO:For the <a href="https://github.com/apache/pinot/pull/10451">backwards incompatible change</a>, this is a
```

Comment:
Will we be making changes in the future to do this?

Reply:
Yes thats right. This is to clean up the old location in a future release. We will lose status of older/completed jobs which should be ok (cc @Jackie-Jiang).

Reply:
is it possible that controller restart/upgrade occurred during a segment reload process is still in progress and can end up in a weird state?

Reply:
Yes, there's that window. We debated about the complexity of making it backwards compatible Vs keeping it simple. If the controller is upgraded after reload is initiated, we will not be able to get the reload status from the upgraded controller (old controller will still be able to read the status).  Given this is a corner case, calling this backwards incompatible can warn users doing a controller upgrade?